starknet_devnet/predeployed_contract_wrapper.py

```diff
@@ -0,0 +1,97 @@
+"""Parent for predeployed contract wrapper classes"""
+from abc import ABC, abstractmethod
+
+from starkware.starknet.testing.starknet import Starknet
+from starkware.starknet.services.api.contract_class import ContractClass
+from starkware.starknet.testing.contract import StarknetContract
+
+
+class PredeployedContractWrapper(ABC):
+    """Parent class for all predeployed contract wrapper classes"""
+
+    # Cannot import it because of circular imports
+    # from .starknet_wrapper import StarknetWrapper
+
+    starknet_wrapper = NotImplemented
+
+    @property
+    def address(self) -> int:
+        try:
+            return self.ADDRESS
+        except AttributeError:
+            try:
+                return self._address
+            except AttributeError:
+                raise NotImplementedError(
+                    "Predeployed contracts should provide an address attribute or an ADDRESS class attribute"
+                )
+
+    @address.setter
+    def address(self, address: int):
+        self._address = address
+
+    @property
+    def class_hash_bytes(self) -> bytes:
+        try:
+            return self._class_hash_bytes
+        except AttributeError:
+            try:
+                return self.HASH_BYTES
+            except AttributeError:
+                raise NotImplementedError(
+                    "Predeployed contracts should provide a class_hash_bytes attribute or a HASH_BYTES class attribute"
+                )
+
+    @class_hash_bytes.setter
+    def class_hash_bytes(self, class_hash_bytes: bytes):
+        self._class_hash_bytes = class_hash_bytes
+
+    @property
+    def contract_class(self) -> ContractClass:
+        try:
+            return self._contract_class
+        except AttributeError:
+            try:
+                return self.get_contract_class()
+            except AttributeError:
+                raise NotImplementedError(
+                    "Predeployed contracts should provide a contract_class attribute or get_contract_class method"
+                )
+
+    @contract_class.setter
+    def contract_class(self, contract_class):
+        self._contract_class = contract_class
+
+    async def mimic_constructor(self):
+        pass
```

Comment:
No. Since not all subclasses needs to override it, ex: UDC

Reply:
I think it should because child classes should be the ones with an empty implementation. The method in the parent class should be abstract (but not by means of `pass`ing).

Also, if `mimic_constructor` is only used inside the class (or its children), it should be prefixed with an underscore (`_mimic_constructor`) to show protectedness.